core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -66,19 +67,22 @@ static std::atomic<int> num_uvm_allocations(0);
 }  // namespace
 
 void DeepCopyCuda(void *dst, const void *src, size_t n) {
-  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaMemcpy(dst, src, n, cudaMemcpyDefault));
+  KOKKOS_IMPL_CUDA_SAFE_CALL((CudaInternal::singleton().cuda_memcpy_wrapper(
+      dst, src, n, cudaMemcpyDefault)));
 }
 
 void DeepCopyAsyncCuda(const Cuda &instance, void *dst, const void *src,
                        size_t n) {
   KOKKOS_IMPL_CUDA_SAFE_CALL(
-      cudaMemcpyAsync(dst, src, n, cudaMemcpyDefault, instance.cuda_stream()));
+      (instance.impl_internal_space_instance()->cuda_memcpy_async_wrapper(
+          dst, src, n, cudaMemcpyDefault, instance.cuda_stream())));
 }
 
 void DeepCopyAsyncCuda(void *dst, const void *src, size_t n) {
   cudaStream_t s = cuda_get_deep_copy_stream();
```

Comment:
@masterleinad I don't understand exactly what the deep copy stream and space are. It looks to be separate from the singleton instance?

So then would the call below `CudaInternal::singleton().cuda_memcpy_async_wrapper(()` still need the string as an argument? Since otherwise it would use the singleton stream.

Reply:
The `deep_copy` stream is managed by the singleton but is not the same as the singleton's stream. So yes, that wrapper would still need a stream argument.